### ISSUE TITLE
Build fails on "cannot find -lpcap"

### ISSUE THREAD

**Issue by mrosi** [rust-pcap/pcap#307]
Hi,
I've installed `libpcap-devel` , but still I'm not able to build, while it fails on:
```
note: /usr/bin/ld: cannot find -lpcap
          collect2: error: ld returned 1 exit status
```
my environment is:
```
[mrosi@developer testcapture]$ ll /usr/lib64/libpcap*
lrwxrwxrwx. 1 root root     12 Aug 10  2021 /usr/lib64/libpcap.so -> libpcap.so.1
lrwxrwxrwx. 1 root root     17 Aug 10  2021 /usr/lib64/libpcap.so.1 -> libpcap.so.1.10.0
-rwxr-xr-x. 1 root root 319688 Aug 10  2021 /usr/lib64/libpcap.so.1.10.0

[mrosi@developer testcapture]$ pkg-config --list-all | grep libpcap
libpcap                        libpcap - Platform-independent network traffic capture library

[mrosi@developer testcapture]$ cat /etc/os-release
NAME="CentOS Stream"
VERSION="9"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="9"
PLATFORM_ID="platform:el9"
PRETTY_NAME="CentOS Stream 9"
ANSI_COLOR="0;31"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:centos:centos:9"
HOME_URL="https://centos.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux 9"
REDHAT_SUPPORT_PRODUCT_VERSION="CentOS Stream"

[mrosi@developer testcapture]$ $ rpm -qa | grep libpcap
libpcap-1.10.0-4.el9.x86_64
libpcap-devel-1.10.0-4.el9.x86_64

[mrosi@developer testcapture]$ cargo --version
cargo 1.72.0 (103a7ff2e 2023-08-15)

[mrosi@developer testcapture]$ rustup --version
rustup 1.26.0 (5af9b9484 2023-04-05)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.72.0 (5680fa18f 2023-08-23)`
```

Thank you for any help

**Comment by stappersg** [comment]
On Wed, Sep 06, 2023 at 12:07:00PM -0700, mrosi wrote:
> Hi,
> I've installed `libpcap-devel` , but still I'm not able to build, while it fails on:
> ```
> note: /usr/bin/ld: cannot find -lpcap
>           collect2: error: ld returned 1 exit status
> ```

```shell
echo $PATH
echo lib
echo $LIBPATH
```

Acutaly:  I liked the **good** bugreport.


Regards
Geert Stappers
Fooly^WFully aware that _github.com_ doesn't render this email.

**Comment by mrosi** [comment]
```
$ echo $PATH
/home/mrosi/.cargo/bin:/home/mrosi/.local/bin:/home/mrosi/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin
```
`$LIBPATH` is not defined

**Comment by stappersg** [comment]
On Wed, Sep 06, 2023 at 12:53:09PM -0700, mrosi wrote:
> `$LIBPATH` is not defined

https://en.wikipedia.org/wiki/Environment_variable has

$LD_LIBRARY_PATH
    On many Unix systems with a dynamic linker, contains a colon-separated
    list of directories that the dynamic linker should search for shared
    objects when building a process image after exec, before searching
    in any other directories.

$LIBPATH or $SHLIB_PATH
    Alternatives to $LD_LIBRARY_PATH typically used on older Unix versions.


Regards
Geert Stappers
Did learn today that LIBPATH got a successor
-- 
Silence is hard to parse

**Comment by mrosi** [comment]
None of them is defined.
Not sure it will help, but I've expanded "note" returned in the error /included above/ and `libpcap-*.rlib` is present in the list and on disk as well

**Comment by Wojtek242** [comment]
@mrosi - not really sure what's happening. My best guess ATM is that the pkg-config code in build.rs is not picking up the location of the library. Can you try setting the environment variable `LIBPCAP_LIBDIR=/usr/lib64/` when compiling? I am not sure if this will work, but if it does then I think I know what's wrong.

**Comment by mrosi** [comment]
@Wojtek242 - I've checked two ways:
1. `export LIBPCAP_LIBDIR=/usr/lib64/`
2. set `LIBPCAP_LIBDIR = "/usr/lib64/"` in `config.toml`

but without any impact

**Comment by Wojtek242** [comment]
That was all the ideas I had. At least from the point of view of this crate. Are you able to link with libpcap in another language? E.g. a simple few line C or C++ project on that same machine/os?

**Comment by Payn** [comment]
@mrosi did you by any chance solve this issue since?

Having the same problem, also on CentOS (7).
Only difference for me is that my pcap version is 1.5.3 and `pkg-config --list-all` doesn't list pcap.
None of the ideas posted here worked for me.

**Comment by mrosi** [comment]
@Payn , no I've moved to other packet manipulation library

**Comment by Wojtek242** [comment]
Closing as stale issue. If somebody else has similar problems, please re-open and comment with details and logs of your problem.
